src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -2424,23 +2548,20 @@ private void truncateBlocking(boolean noSnapshot)
         final long truncatedAt;
         final CommitLogPosition replayAfter;
 
-        if (!noSnapshot && (keyspace.getMetadata().params.durableWrites || DatabaseDescriptor.isAutoSnapshot()))
+        if (!noSnapshot &&
+               (keyspace.getMetadata().params.durableWrites && !memtableWritesAreDurable())  // need to clear dirty regions
+               || DatabaseDescriptor.isAutoSnapshot()) // need sstable for snapshot
```

Comment:
The previous logic was equivalent to...
```
if (!noSnapshot && keyspace.getMetadata().params.durableWrites || !noSnapshot && DatabaseDescriptor.isAutoSnapshot()))
```
...but now the `if` condition is true if `DatabaseDescriptor.isAutoSnapshot() == true` no matter what the value of `noSnapshot` is. (ex. If `noSnapshot == true`, we'll still force a blocking flush if `isAutoSnapshot() == true`.) Is that intended?

Reply:
Not at all, corrected.